### PR TITLE
many additions in here - perhaps most impactful - #261 regarding #260

### PR DESCRIPTION
**Pull request by goodb**
Key updates here:
1. minerva will add root entity classes as types for gene products #260 this will allow it to operate with or without the presence of all entities classes (neo) in the go-lego.owl file.  The intention is to take neo out of go-lego.  
2. added a search API used in Tremayne's model search tool #230 
3. shex validation code in server and on the command line #212 
4. owl validation code for command line #254

### CODE REVIEW

**Comment by kltm** [comment]
Noting that all functionality is already in use on noctua-dev--there is nothing "new" here.
Talking to @goodb, as far as is known, there are no incompatible changes--this should be a drop-in replacement.
If @balhoff has no issues, I'll merge. (It may take a little longer to get the code live.)

**Comment by kltm** [comment]
@goodb Apparently minerva-cli/src/main/java/org/geneontology/minerva/cli/MinervaCommandRunner.java  is conflicting?

**Comment by goodb** [comment]
That file has been deleted in the current dev build.

**Comment by goodb** [comment]
What is the right pattern to resolve?  I guess deleting it directly from master ought to work, right?

**Comment by kltm** [comment]
Assuming that it's an unused file on `master`, I'd just delete. I'm not sure why that would be considered "conflicting" however.

**Comment by goodb** [comment]
I don't understand the conflict either.  On the current master build it is used.  (It just wraps the command line client code - I saw no need for two classes when I redid the cmdline).

**Comment by kltm** [comment]
Well, perhaps a little adventure then after the merge (pending @balhoff ) then. Worst case is we roll back, so no worries.

**Comment by goodb** [comment]
Hmm..  I think I just made a mistake.  I deleted it from master..  rendering the current master build broken, but this pull request mergeable....  That seems bad in the case we have anything build from master prior to the merge of this pr..

**Comment by goodb** [comment]
reverted master to a functional version.   When ready to merge, safe to delete offender...

File: minerva-lookup/src/main/java/org/bbop/golr/java/RetrieveGolrOntologyClass.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		public String source;
+		public String is_obsolete;
+		public List<String> alternate_id;
+		public List<String> replaced_by;
+		public List<String> consider;
+		public List<String> synonym;
+		public List<String> subset;
+		public List<String> definition_xref;
+		public List<String> database_xref;
+		public List<String> isa_partof_closure;
```
Comment (balhoff) [inline minerva-lookup/src/main/java/org/bbop/golr/java/RetrieveGolrOntologyClass.java]: Is it possible to have just an `isa_closure`?

Comment (balhoff) [reply minerva-lookup/src/main/java/org/bbop/golr/java/RetrieveGolrOntologyClass.java]: I left a couple of other comments on https://github.com/geneontology/minerva/pull/261/files

File: minerva-server/src/main/resources/GetAllModels.rq [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  GRAPH ?cam {      
+      ?cam <http://purl.org/dc/elements/1.1/title> ?title ;
+           <http://purl.org/dc/elements/1.1/date> ?date ;
+           <http://purl.org/dc/elements/1.1/contributor> ?contributor ;
+           lego:modelstate ?state .          
+    	   optional { 
+             ?cam <http://www.geneontology.org/formats/oboInOwl#id> ?id 
+           }       	
+           #lpalbou: Baby Proofing the query since oboInOwl#id is not always there
+	  		BIND(IF(bound(?id), ?id, concat("gomodel:", substr(str(?cam), 31))) as ?id) .
```
Comment (balhoff) [inline minerva-server/src/main/resources/GetAllModels.rq]: It seems like turning an IRI into a CURIE should happen somewhere else, not here.

File: minerva-server/src/main/resources/ModelSearchQueryTemplate.rq [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+SELECT  <return_block>    
+WHERE {
+  GRAPH ?cam {  
+        ?cam <http://purl.org/dc/elements/1.1/title> ?title ;
+           <http://purl.org/dc/elements/1.1/date> ?date ;
+           <http://purl.org/dc/elements/1.1/contributor> ?contributor ;   
+        optional{?cam <http://purl.org/pav/providedBy> ?group } .   
+        optional{?cam lego:modelstate ?state } . 
+	  BIND(IF(bound(?id), ?id, concat("gomodel:", substr(str(?cam), 31))) as ?id) .
```
Comment (balhoff) [inline minerva-server/src/main/resources/ModelSearchQueryTemplate.rq]: We should have one place to shorten IRIs.

**Comment by goodb** [comment]
@kltm I think we are good to go here for dev testing.  Fixed problems @balhoff identified.

**Comment by kltm** [comment]
@goodb GH is still identifying one change from @balhoff --are we skipping that?

File: minerva-lookup/src/main/java/org/bbop/golr/java/RetrieveGolrOntologyClass.java (hunk 1 above)
Comment (goodb) [reply minerva-lookup/src/main/java/org/bbop/golr/java/RetrieveGolrOntologyClass.java]: fixed in last merge into dev

File: minerva-server/src/main/resources/GetAllModels.rq (hunk 2 above)
Comment (goodb) [reply minerva-server/src/main/resources/GetAllModels.rq]: yup fixed

**Comment by goodb** [comment]
@kltm I think he just didn't officially resolve them.  He approved the PR that fixed everything.  May have confused it as I made two branches on the way.  My understanding is that we are thumbs up from @balhoff .

**Comment by balhoff** [comment]
👍

**Comment by goodb** [comment]
@kltm FYI I am ready to run model updater.

**Comment by kltm** [comment]
This is to inject the entity types into the models for https://github.com/geneontology/minerva/issues/260 , right? Step two there is now slated for this Friday, so we're a little bit away from that for the moment. We can talk more about that on the call tomorrow if you'd like.

**Comment by goodb** [comment]
Yup.  Its available now from the minerva command line in case some one else wants to run it.  https://github.com/geneontology/minerva/pull/265

**Comment by kltm** [comment]
@goodb Reading through these comments again, it looks like you attempted to fix the conflict, but were unable to and reverted the attempt. Could you attempt the merge locally and see what is actually going on first?

**Comment by goodb** [comment]
Deleting the file on master fixes the problem and makes merge possible.  Not sure why it is a conflict.  Sorry, off computer right now.

**Comment by kltm** [comment]
@goodb I tried that and seem to get compilation problems (in gitter). I think we need to just punt on this for now--I'm having separate issues with the new models on `dev` apparently.

**Comment by goodb** [comment]
Master will produce compilation problems without that file.  That’s why I reverted the change while waiting for dev merge.  If you delete it on master and then merge in from dev , it should work.

**Comment by kltm** [comment]
@goodb Merge something in some state anyways. We shall see.

**Pull request merged**